File: packetCapture/src/main/java/com/ninpou/packetcapture/core/tunnel/RawTcpTunnel.java
package com.ninpou.packetcapture.core.tunnel;

import java.io.IOException;
import java.net.InetSocketAddress;
import java.nio.ByteBuffer;
import java.nio.channels.Selector;
import java.nio.channels.SocketChannel;

public class RawTcpTunnel extends BaseTcpTunnel {

    public RawTcpTunnel(SocketChannel innerChannel, Selector selector) {
        super(innerChannel, selector);
    }

    public RawTcpTunnel(InetSocketAddress serverAddress, Selector selector, short portKey) throws IOException {
        super(serverAddress, selector, portKey);
    }

    @Override
    protected void onConnected() throws Exception {
        onTunnelEstablished();
    }

    @Override
    protected boolean isTunnelEstablished() {
        return true;
    }

    @Override
    protected void beforeSend(ByteBuffer buffer) throws Exception {
    }

    @Override
    protected void afterReceived(ByteBuffer buffer) throws Exception {
    }

    @Override
    protected void onDispose() {
    }
}

File: packetCapture/src/main/java/com/ninpou/packetcapture/core/tunnel/RawTcpTunnel.kt
package com.ninpou.packetcapture.core.tunnel

import java.net.InetSocketAddress
import java.nio.ByteBuffer
import java.nio.channels.Selector
import java.nio.channels.SocketChannel

open class RawTcpTunnel : BaseTcpTunnel {
    constructor(innerChannel: SocketChannel?, selector: Selector?) : super(innerChannel, selector) {}
    constructor(serverAddress: InetSocketAddress?, selector: Selector?, portKey: Short) : super(serverAddress, selector, portKey) {}

    @Throws(Exception::class)
    override fun onConnected() {
        onTunnelEstablished()
    }

    override val isTunnelEstablished: Boolean
        protected get() = true

    @Throws(Exception::class)
    override fun beforeSend(buffer: ByteBuffer?) {
    }

    @Throws(Exception::class)
    override fun afterReceived(buffer: ByteBuffer?) {
    }

    override fun onDispose() {}
}